arrow-libs/core/arrow-autoclose/src/jvmMain/kotlin/arrow/throwIfFatal.kt

```diff
@@ -1,10 +1,8 @@
 package arrow
 
-import kotlin.coroutines.cancellation.CancellationException
-
 @PublishedApi
 internal actual fun Throwable.throwIfFatal(): Throwable =
   when(this) {
-    is VirtualMachineError, is ThreadDeath, is InterruptedException, is LinkageError, is CancellationException -> throw this
+    is VirtualMachineError, is ThreadDeath, is InterruptedException, is LinkageError -> throw this
```

Comment:
I understand in some places it's desirable to handle CancellationException separately but it's confusing to have two definitions of "fatal" exceptions (this vs `arrow.core.NonFatal`)
It may be clearer to explicitly handle CancellationException where required and use the `NonFatal` in failure cases.

Reply:
That's fair, I'll add an explicit carve-out for CancellationException instead

Reply:
Done!